backends/dpdk/dpdkUtils.cpp

```diff
@@ -0,0 +1,56 @@
+/* Copyright 2022 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "dpdkUtils.h"
+
+namespace DPDK {
+bool isSimpleExpression(const IR::Expression *e) {
+    if (e->is<IR::Member>() || e->is<IR::PathExpression>() ||
+        e->is<IR::Constant>() || e->is<IR::BoolLiteral>())
+        return true;
+    return false;
+}
+
+bool isNonConstantSimpleExpression(const IR::Expression *e) {
+    if (e->is<IR::Member>() || e->is<IR::PathExpression>())
+        return true;
+    return false;
+}
+
+bool isCommutativeBinaryOperation(const IR::Operation_Binary *bin) {
+    auto right = bin->right;
+    if (right->is<IR::Add>() || right->is<IR::Equ>() || right->is<IR::LOr>() ||
+        right->is<IR::LAnd>() || right->is<IR::BOr>() || right->is<IR::BAnd>() ||
+        right->is<IR::BXor>())
+        return true;
+    return false;
+}
+
+bool isStandardMetadata(cstring name) {
```

Comment:
This function is actually PSA-specific, not DPDK specific. So it should probably move to some common file eventually.

Reply:
How about creating a utils.h/cpp under backends/common/ ?

Reply:
We should probably move more stuff in such a place. But it should be in some directory named "psa".

Reply:
This specific function is for both psa and pna metadata. Of course, we can split it up if we decide to go with arch-specific common utils.
Can this PR be merged if this does not need to be an immediate action?